genshin/utility/logfile.py

```diff
@@ -31,16 +50,29 @@ def _read_logfile(logfile: typing.Optional[PathLike] = None) -> str:
 
     logfile = logfile or get_logfile()
     if logfile is None:
-        raise FileNotFoundError("No Genshin Installation was found, could not get gacha data.")
+        raise FileNotFoundError(
+            "No Genshin Installation was found, could not get gacha data."
+            "Pleas set game location using genshin.utility.logfile.set_geme_location(game location)"
+        )
 
-    return logfile.read_text()
+    try:
+        # won't work if genshin is running or script using this function is run as administrator
+        return logfile.read_text(errors='replace')
+    except PermissionError as ex:
+        raise PermissionError(
+            "Pleas turn off genshin impact or try running script as administrator!"
+        ) from ex
 
 
 def extract_authkey(string: str) -> typing.Optional[str]:
     """Extract an authkey from the provided string."""
-    match = re.search(r"https://.+?authkey=([^&#]+)", string, re.MULTILINE)
+    match = re.search(
+        r"https://webstatic-sea.hoyoverse.com/genshin/event/e20190909gacha-v2.+?authkey=([^&#]+)",
```

Comment:
No need to enforce the precise url.

Reply:
Actually there is. Before I added this i didn't work for me

Reply:
Maybe at least make it work for both history and details.